memory/combat_manager.py

```diff
@@ -301,28 +343,44 @@ def _read_players(self):
                                 0x0,
                             ],
                         )
-                    except Exception:
-                        return
 
                     # This check was added due to the pointer not falling off in time, referencing
                     # an enemy that just died
-                    if (
-                        target_unique_id_base is self.NULL_POINTER
-                        or target_unique_id_base is None
-                    ):
-                        continue
-
-                    selected_target_guid = self.memory.read_guid(
-                        target_unique_id_base + 0x14
-                    )
+                    try:
+                        selected_target_guid = self.memory.read_guid(
+                            target_unique_id_base + 0x14
+                        )
+                    except Exception:
+                        selected_target_guid = ""
 
                     mp_text_field = self.memory.follow_pointer(item, [0x30, 0x0])
                     current_mp = self.memory.read_int(mp_text_field + 0x54)
+
+                    # if the current player is selected, set it to the main combat manager state
+                    # this will help us prevent scanning lists later on
+                    if selected:
+                        selected_character = character
+
                     player = CombatPlayer()
+
+                    # TODO: hardcode these for now - we need to extract these players into
```

Comment:
Worst case, we can keep them in a dict somewhere. I assume that this stat is actually dependent on level and levelup choices as well.

Reply:
Agreed - this is more for "just testing" because the game doesn't expose stats, so we'll have to build some solution to find the equipped items, and add their values to the base stats...